src/fheroes2/world/world_loadmap.cpp

```diff
@@ -1591,7 +1591,14 @@ void World::setUltimateArtifact()
         assert( radius >= 0 );
 
         // Remove the predefined Ultimate Artifact object.
-        existingUltimateArtIter->removeObjectPartsByUID( existingUltimateArtIter->getMainObjectPart()._uid );
+        const uint32_t uid = existingUltimateArtIter->getMainObjectPart()._uid;
+        existingUltimateArtIter->removeObjectPartsByUID( uid );
+
+        // Also, remove the shadow of the Artifact.
+        const int32_t neighborTileIndex = Maps::GetDirectionIndex( existingUltimateArtIter->GetIndex(), Direction::LEFT );
+        if ( neighborTileIndex >= 0 ) {
+            vec_tiles[neighborTileIndex].removeObjectPartsByUID( uid );
+        }
```

Comment:
We can slightly improve the logic: check if direction is valid (not only `index >= 0`) and use `tileId` that has the value of `existingUltimateArtIter->GetIndex()`:
```suggestion
        if ( Maps::isValidDirection( tileId, Direction::LEFT ) ) {
            const int32_t neighborTileIndex = Maps::GetDirectionIndex( tileId, Direction::LEFT );
            vec_tiles[neighborTileIndex].removeObjectPartsByUID( uid );
        }
```
What do you think?